go-controller/pkg/libovsdbops/template_var.go

```diff
@@ -0,0 +1,231 @@
+	for _, nbTemplate := range flattened {
+		callback(nbTemplate)
+	}
+}
+
+// ListTemplates looks up all chassis template variables. It returns
+// libovsdb.Templates values indexed by template name.
+func ListTemplates(nbClient libovsdbclient.Client) (templatesByName TemplateMap, err error) {
+	templatesByName = TemplateMap{}
+	err = nil
+	ctx, cancel := context.WithTimeout(context.Background(), types.OVSDBTimeout)
+	defer cancel()
+
+	templatesList := []*nbdb.ChassisTemplateVar{}
+	err = nbClient.List(ctx, &templatesList)
+	if err != nil {
+		return
+	}
+
+	for _, nbTemplate := range templatesList {
+		for name, perChassisValue := range nbTemplate.Variables {
+			tv, found := templatesByName[name]
+			if !found {
+				tv = MakeTemplate(name)
+				templatesByName[name] = tv
+			}
+			tv.Value[nbTemplate.Chassis] = perChassisValue
+		}
+	}
+	return
+}
+
+func mutateChassisTemplateVarOps(nbClient libovsdbclient.Client, preserveVarValues bool,
+	ops []libovsdb.Operation, templateVars ...TemplateMap) ([]libovsdb.Operation, error) {
+	opModels := []operationModel{}
+	forEachNBTemplateInMaps(templateVars, func(nbTemplate *nbdb.ChassisTemplateVar) {
+		opModels = append(opModels, operationModel{
+			Model: nbTemplate,
+			ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+				return item.Chassis == nbTemplate.Chassis
+			},
+			OnModelMutations:                  []interface{}{&nbTemplate.Variables},
+			ErrNotFound:                       false,
+			BulkOp:                            false,
+			PreserveExistingMapValuesOnMutate: preserveVarValues,
+		})
+	})
+
+	modelClient := newModelClient(nbClient)
+	return modelClient.CreateOrUpdateOps(ops, opModels...)
+}
+
+func CreateChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars ...TemplateMap) ([]libovsdb.Operation, error) {
+
+	return mutateChassisTemplateVarOps(nbClient, true, ops, templateVars...)
+}
+
+func CreateOrUpdateChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars ...TemplateMap) ([]libovsdb.Operation, error) {
+
+	return mutateChassisTemplateVarOps(nbClient, false, ops, templateVars...)
+}
+
+func DeleteChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars ...TemplateMap) ([]libovsdb.Operation, error) {
+	opModels := []operationModel{}
+	forEachNBTemplateInMaps(templateVars, func(nbTemplate *nbdb.ChassisTemplateVar) {
+		deleteNbTemplate := &nbdb.ChassisTemplateVar{
+			Chassis:   nbTemplate.Chassis,
+			Variables: map[string]string{},
+		}
+		for name := range nbTemplate.Variables {
+			deleteNbTemplate.Variables[name] = ""
+		}
+		opModels = append(opModels, operationModel{
+			Model: deleteNbTemplate,
+			ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+				return item.Chassis == deleteNbTemplate.Chassis
+			},
+			OnModelMutations: []interface{}{&deleteNbTemplate.Variables},
+			ErrNotFound:      false,
+			BulkOp:           false,
```

Comment:
I think not.  The `ModelPredicate` for each mutation ensures that only one model (the one matching the Chassis name) is updated.  The chassis name is an index of "Chassis_Template_Var" in the northbound database so that ensures uniqueness AFAICT.